File: android/DFM/src/main/java/com/dontflymoney/auth/Authentication.java
package com.dontflymoney.auth;

import android.content.Context;

import com.dontflymoney.io.File;
import com.dontflymoney.io.FileNames;

public class Authentication
{
	private Context context;
	private SafeTicket safe;
	
	public Authentication(Context context)
	{
		this.context = context;
		safe = new SafeTicket(this.context);
	}
	
	public void Set(String ticket)
	{
		File ticketFile = new File(context, FileNames.Ticket);
		
		String encryptedTicket = safe.Encrypt(ticket);
		
		ticketFile.WriteToFile(encryptedTicket);
	}
	
	public String Get()
	{
		File ticketFile = new File(context, FileNames.Ticket);
		
		String encryptedTicket = ticketFile.ReadFromFile();
		
		return safe.Decrypt(encryptedTicket);
	}

	public boolean IsLoggedIn()
	{
		String ticket = Get();

		return ticket != null && !ticket.isEmpty();
	}

	public void Clear()
	{
		Set(null);
	}
	

}

File: android/DFM/src/main/kotlin/com.darakeon.dfm/auth/Authentication.kt
package com.darakeon.dfm.auth

import android.content.Context
import android.os.Build

class Authentication(private val context: Context) {
	private val aks =
		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M)
			Aks()
		else
			OldAks(context)

	private val spKey = "Ticket"

	var ticket: String
		get() {
			val encryptedTicket = context.getValue(spKey)
			return aks.decrypt(encryptedTicket)
		}
		set(value) {
			val encryptedTicket = aks.encrypt(value)
			context.setValue(spKey, encryptedTicket)
		}

	fun clear() {
		ticket = ""
	}
}
